examples/example_mc_fit_a_spectrum.py

```diff
@@ -19,7 +19,11 @@
         resource_filename("amespahdbpythonsuite", "resources/galaxy_spec.ipac")
     )
 
-    # Convert spectral unit to wavenumber
+    # Store units for plotting.
```

Comment:
Passing the units explicitly should not be necessary as the mcfitted class holds a copy of the unperturbed observation from where the units can be found.

Reply:
The problem is that that we do `obs.abscissaunitsto("1/cm")` but when plotting the units should be in microns.

Reply:
The fit is done in wavenumber-space, thus it is the 'natural' unit for the class to use for plotting. The wavelength-keyword can be set for plotting using a wavelength spectral axis.

Reply:
The wavelength keyword is set when plotting, but it does not hold the units. The units, currently, has to be provided and the ones stored after `obs.abscissaunitsto("1/cm")` are 1/cm.

Reply:
The concept here is 'object-encapsulation'; the data in a class-object should be self-contained, i.e., units should not be needed to be provided externally. When plotting you call the a method on the mcfitted-class 'to plot itself', which has internally the data stored in 1/cm units. For convenience, a keyword can be provided to use um units instead.

Reply:
The input spectrum units has to be stored before changed and carried over to the fitted object to be able to call it.

Reply:
Looking into the returned fitted object, the units key or even the observation or getobservation methods does not retain the original units. It doesn't seem to have any property that holds those and this means this has to be implemented and returned in the observation.py method.

Reply:
No. What do you mean by 'original' units? The original units of the _fit_ for the spectral axis are 1/cm, which is what the stored Spectrum1D object has. That these units might not be those that were read in initially from file doesn't matter here as everything is now in the the context of the mcfitted-object, i.e., spectra are fitted in wavenumber-space. This is what 'object-encapsulation' and 'ownership' are all about in programming: mcfitted owns a copy of observations in the units it uses. Everything in the object/class is self-contained/consistent, i.e., you should not need external information to plot the results. **TLDR**; The units of the spectral axes within the mcfitted-class should be 1/cm, including that of the unperturbed Spectrum1D-object stored in the property observation. For convenience, the wavelength-keyword can be used to plot the fit results against micron, which, again, are done in wavenumber-space.

Reply:
I understand that. I was thinking in a broader context where the input spectral units could be anything, not necessarily microns, and the plot would return the result in those units. In any case, currently when the wavelength keyword is given no units are defined and this needs to be fixed, so microns will be the default in that case.

Reply:
No, I don't think you understand. The default units are 1/cm, as that is what the units of the spectral axis are of the data in the database. The observations need to be changed to match those in the database and is why there is an error when providing other units, i.e., µm. The observation-object stored in an mcfitted-object is **not** the one that is read-in from file; an internal copy/translation is made into a Spectrum1D-object. It is the user's responsibility to provide the observations in the correct units of 1/cm. If no units are provided for the spectral axis by the user, the fit is still performed, but it is assumed the units of the spectral axis are 1/cm: see https://github.com/PAHdb/AmesPAHdbPythonSuite/blob/19abd4ce8fdf384833d67ba89bde98e6a31d56c8/amespahdbpythonsuite/spectrum.py#L417, where the database units are set for the spectral axis of the created Spectrum1D-object.

Reply:
Well, as I said, I do understand both how the fitting is performed as well as the units in the database. However, you said  "It is the user's responsibility to provide the observations in the correct units", but this is mentioned nowhere in the documentation. Therefore, as it is now, the user can provide any units, which typically are microns, and of course these have to be converted.

Reply:
Again, no. If the user provides any units other than 1/cm an error messages is printed (raising an exception would be better here): https://github.com/PAHdb/AmesPAHdbPythonSuite/blob/19abd4ce8fdf384833d67ba89bde98e6a31d56c8/amespahdbpythonsuite/spectrum.py#L424.

Reply:
That is true, but we need to state this in the documentation as well and perhaps expand the comment in the example, as the user should not have to find out and deal with this after running the code.

Reply:
Yes, you're certainly correct there. The documentation needs a lot of TLC. One place to document this clearly is in the method preamble, which used by sphinx to generate the API-docs. This discussion was very meaningful, as I went back to check the code and it appears actually no copy is being made of the input! I'll try and fix that ...

Reply:
Please wait a few more minutes to make the PR to avoid rebasing! :)
I am adjusting the tests now.

Reply:
Of course.